patches/proxy-uri.diff

```diff
@@ -98,3 +98,66 @@ Index: code-server/lib/vscode/src/vs/workbench/contrib/terminal/common/terminalE
  
  		// Merge config (settings) and ShellLaunchConfig environments
  		mergeEnvironments(env, allowedEnvFromConfig);
+Index: code-server/lib/vscode/src/vs/code/browser/workbench/workbench.ts
+===================================================================
+--- code-server.orig/lib/vscode/src/vs/code/browser/workbench/workbench.ts
++++ code-server/lib/vscode/src/vs/code/browser/workbench/workbench.ts
+@@ -17,6 +17,7 @@ import { isFolderToOpen, isWorkspaceToOp
+ import { create, ICredentialsProvider, IURLCallbackProvider, IWorkbenchConstructionOptions, IWorkspace, IWorkspaceProvider } from 'vs/workbench/workbench.web.main';
+ import { posix } from 'vs/base/common/path';
+ import { ltrim } from 'vs/base/common/strings';
++import { relativeRoot } from 'vs/server/node/webClientServer';
+ 
+ interface ICredential {
+ 	service: string;
+@@ -485,6 +486,7 @@ function doCreateUri(path: string, query
+ 		});
+ 	}
+ 
++	path = (window.location.pathname + "/" + path).replace(/\/\/+/g, "/")
+ 	return URI.parse(window.location.href).with({ path, query });
+ }
+ 
+@@ -496,7 +498,7 @@ function doCreateUri(path: string, query
+ 	if (!configElement || !configElementAttribute) {
+ 		throw new Error('Missing web configuration element');
+ 	}
+-	const config: IWorkbenchConstructionOptions & { folderUri?: UriComponents; workspaceUri?: UriComponents; callbackRoute: string } = JSON.parse(configElementAttribute);
++	const config: IWorkbenchConstructionOptions & { folderUri?: UriComponents; workspaceUri?: UriComponents; callbackRoute: string } = { ...JSON.parse(configElementAttribute), remoteAuthority: location.host }
+ 
+ 	// Create workbench
+ 	create(document.body, {
+@@ -506,6 +508,32 @@ function doCreateUri(path: string, query
+ 		} : undefined,
+ 		workspaceProvider: WorkspaceProvider.create(config),
+ 		urlCallbackProvider: new LocalStorageURLCallbackProvider(config.callbackRoute),
+-		credentialsProvider: config.remoteAuthority ? undefined : new LocalStorageCredentialsProvider() // with a remote, we don't use a local credentials provider
++		credentialsProvider: config.remoteAuthority ? undefined : new LocalStorageCredentialsProvider(), // with a remote, we don't use a local credentials provider
++		resolveExternalUri: (uri: URI): Promise<URI> => {
++			// TODO@jsjoeio - how do we make it relative to work when hosted on subpath?
++			const baseUrl = `${window.location.protocol}//${window.location.host}`
```

Comment:
Wondering if there's a better way to determine the `baseUrl`. `relativeRoot` is something we use but it's not available on the browser side AFAIK.

Reply:
We need to resolve against the path in case code-server is hosted at a sub-path which means we have to get `proxyEndpointTemplate` in here somehow as it contains the information needed to resolve relatively.  Then we can do:

```
resolved = new URL(proxyEndpointTemplate.replace('{port}', port), window.location.href)
```

Reply:
Super helpful! Thank you!